.github/workflows/shopify-deploy.yml

```diff
@@ -0,0 +1,147 @@
+name: Naturally Sweet Shopify Theme Deployment
+
+on:
+  push:
```

Comment:
My GHA might be dated, but I found this too sensitive in the past.

I have previously used the `pull_request` trigger instead and in the Job added a merge check to only run the job when the merge completed.


```
on:
  pull_request:
    types:
      - closed
    branches:
      - staging

jobs:
  theme-check:
    name: Theme Check
    # PR must must be merged, not closed to trigger a deployment
    if: github.event.pull_request.merged
   ...
```